### ISSUE TITLE
Miss BC_IMPOSSIBLE_DOWNCAST Warning for Invalid HashMap to LinkedHashMap Cast​

### ISSUE THREAD

**Issue by stoneman258** [spotbugs/spotbugs#3788]
SpotBugs fails to report a `BC_IMPOSSIBLE_DOWNCAST` warning when encountering an invalid downcast from `HashMap` to `LinkedHashMap` in the provided code example (line 6). The code violates Java type safety rules, which will trigger a ClassCastException during program execution. So, thisi s a false negative bug.

### Code Example
```java
import java.util.HashMap;
import java.util.LinkedHashMap;

public class Main {
    private static LinkedHashMap foo(HashMap map) {
        return (LinkedHashMap) map; // should report a BC_IMPOSSIBLE_DOWNCAST warning, but no warnings
    }

    public static void main(String args[]) {
        HashMap m = new HashMap();
        System.out.println(m.equals(null));
        LinkedHashMap lhm = foo(m);
    }
}
```

**Comment by JuditKnoll** [comment]
This is a duplicate of https://github.com/spotbugs/spotbugs/issues/3769

**Comment by JuditKnoll** [comment]
In this case `BC_UNCONFIRMED_CAST` is reported in the `foo()` function. However in this simple edge case, `BC_IMPOSSIBLE_DOWNCAST` would be more specific, but  `BC_UNCONFIRMED_CAST` quite similar with the same root cause (cast issues). The more specific report would need an unnecessary complicated detector and less consistent reports.
